### PR TITLE
Update to our latest conventions 

### PR DESCRIPTION
**Pull request by jcredding**
#### Update to our latest conventions

This is just to keep up with the latest conventions and keep the
gem fresh.

#### Update to use latest Assert conventions

This updates the existing tests to use the latest Assert
conventions. This includes using `assert_that` and using `let`s.

#### Switch to `MuchMixin` instead of `MuchPlugin`

This switches to using `MuchMixin` instead of `MuchPlugin`. This
is part of updating to our latest conventions.

### CODE REVIEW

File: .ruby-version [hunk 1]
```diff
@@ -0,0 +1 @@
+2.7.2
```
Comment (kellyredding) [inline .ruby-version]: can we prefer `2.5.8` for gem development please?

File: .t.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+- default_cmd: "bundle exec assert"
+  test_dir: "test"
+  test_file_suffixes:
+    - "_tests.rb"
```
Comment (kellyredding) [inline .t.yml:4]: did it work? Thanks for adding this in!

**Review by kellyredding — APPROVED** [review]
@jcredding awesome!

File: .t.yml (hunk 2 above)
Comment (jcredding) [reply .t.yml:4]: @kellyredding - Works great!

**Pull request merged**